draft-ietf-cats-usecases-requirements.md

```diff
@@ -876,20 +921,18 @@ informative:
    others {{TR22.874}}.  Of course, specific metrics might not be utilized
    at all in other scenarios.  Hence:
 
-   R8:
-   : There MUST exist flexibility in term of metrics definition and
-   utilization for the selection of service instance.
+   R12: In addition to common metrics that are agreed by all CATS
+   components like processing delay, there SHOULD be some ways for
+   metrics definition in different dimensions, which is used for the
```

Comment:
not sure what is meant by "dimensions" here.

Reply:
```suggestion
   allowing the customized metrics definition, which is used for the
```

Reply:
Vendor specific instead of customized? From my understanding, the original text tried to explain there were different ways to express metrics. Some can be standardized without much controversies (called common metrics in the given text), and some may not.

Reply:
BTW, an approach where 'vendor-specific- would work is to define an interface of conveying metrics to the decision algorithms that does not, well, require ANY metric knowledge at all. My suggestion to use mere cardinals at the interface allows for interpreting those cardinals in any way you want at the application and vendor level. You may now have energy-aware decision (e.g., the cardinal being a Joule expression of your instance), compute-aware decision (e.g., the cardinal reflecting a deployed compute unit or even an instantaneous compute load), or a storage-aware decision (e.g., the cardinal expressing the occupancy of the associated hard disk). It can be whatever the application and vendor wants while the interface, conveying just a cardinal, remains the same. But having an interface where we allow for agreed/common metrics and proprietary ones at the same time creeps the vendor lock into the backdoor

Reply:
> BTW, an approach where 'vendor-specific- would work is to define an interface of conveying metrics to the decision algorithms that does not, well, require ANY metric knowledge at all. My suggestion to use mere cardinals at the interface allows for interpreting those cardinals in any way you want at the application and vendor level. You may now have energy-aware decision (e.g., the cardinal being a Joule expression of your instance), compute-aware decision (e.g., the cardinal reflecting a deployed compute unit or even an instantaneous compute load), or a storage-aware decision (e.g., the cardinal expressing the occupancy of the associated hard disk). It can be whatever the application and vendor wants while the interface, conveying just a cardinal, remains the same. But having an interface where we allow for agreed/common metrics and proprietary ones at the same time creeps the vendor lock into the backdoor

What is the difference? You still get proprietary interpretation of the same number which make it hard to interop, right?